Fails to build with GHC 9.2.1

## Problem
hindent fails to build with the new 9.2.1 GHC

- [X] I checked the issue tracker for existing issues about this problem.
- [X] I'm using the latest version of hindent.

output of the error with `cabal install hindent`

```
Failed to build hindent-5.3.2.
Build log (
/home/arjun/.cabal/logs/ghc-9.2.1/hindent-5.3.2-de470b0284a3a0b7cfe4133b121ae750901eb61e4d57dc19c1c629ac31d49982.log
):
Configuring library for hindent-5.3.2..
Preprocessing library for hindent-5.3.2..
Building library for hindent-5.3.2..
[1 of 5] Compiling HIndent.CabalFile ( src/HIndent/CabalFile.hs, dist/build/HIndent/CabalFile.o, dist/build/HIndent/CabalFile.dyn_o )

src/HIndent/CabalFile.hs:50:12: error:
    • Couldn't match type: [Char]
                     with: Distribution.Utils.Path.SymbolicPath
                             Distribution.Utils.Path.PackageDir
                             Distribution.Utils.Path.SourceDir
      Expected: Distribution.Utils.Path.SymbolicPath
                  Distribution.Utils.Path.PackageDir
                  Distribution.Utils.Path.SourceDir
                -> Bool
        Actual: FilePath -> Bool
    • In the first argument of ‘any’, namely ‘inDir’
      In the first argument of ‘($)’, namely ‘any inDir’
      In the expression: any inDir $ hsSourceDirs bi
   |
50 |     in any inDir $ hsSourceDirs bi
   |            ^^^^^
cabal: Failed to build hindent-5.3.2 (which is required by exe:hindent from
hindent-5.3.2). See the build log above for details.
```

--- a/hindent.cabal
+++ b/hindent.cabal
@@ -1,5 +1,5 @@
 name:                hindent
-version:             5.3.2
+version:             5.3.3
 synopsis:            Extensible Haskell pretty printer
 description:         Extensible Haskell pretty printer. Both a library and an executable.
                      .

--- a/src/HIndent/CabalFile.hs
+++ b/src/HIndent/CabalFile.hs
@@ -12,6 +12,9 @@ import Data.Traversable
 import Distribution.ModuleName
 import Distribution.PackageDescription
 import Distribution.PackageDescription.Configuration
+#if MIN_VERSION_Cabal(3, 6, 0)
+import Distribution.Utils.Path (getSymbolicPath)
+#endif
 #if MIN_VERSION_Cabal(2, 2, 0)
 import Distribution.PackageDescription.Parsec
 #else
@@ -47,7 +50,13 @@ mkStanza bi mnames fpaths =
         Just relpath ->
           any (equalFilePath $ dropExtension relpath) modpaths ||
           any (equalFilePath relpath) fpaths
-    in any inDir $ hsSourceDirs bi
+    in any inDir $ hsSourceDirs' bi
+      where
+#if MIN_VERSION_Cabal(3, 6, 0)
+        hsSourceDirs' =  (map getSymbolicPath) . hsSourceDirs
+#else
+        hsSourceDirs' = hsSourceDirs
+#endif
 
 -- | Extract `Stanza`s from a package
 packageStanzas :: PackageDescription -> [Stanza]

